src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
+      dfltLayer.type = 'KML';
+      return dfltLayer;
+    };
+
+    // Creates a new help tooltip
+    var createHelpTooltip = function() {
+      var tooltipElement = $document[0].createElement('div');
+      tooltipElement.className = 'ga-draw-help';
+      var tooltip = new ol.Overlay({
+        element: tooltipElement,
+        offset: [15, 15],
+        positioning: 'top-left',
+        stopEvent: false
+      });
+      return tooltip;
+    };
+
+    // Display an help tooltip for when drawing
+    var updateHelpTooltip = function(overlay, type, drawStarted, onFirstPoint,
+        onLastPoint) {
+      if (!overlay) {
+        return;
+      }
+      var helpMsgId = 'draw_start_';
+      if (drawStarted) {
+        if (type != 'marker' && type != 'annotation') {
+          helpMsgId = 'draw_next_';
         }
-        gaMeasure.registerOverlaysEvents(map, dfltLayer);
-        dfltLayer.label = $translate.instant('draw_layer_label');
-        dfltLayer.type = 'KML';
-        return dfltLayer;
-      };
-      // Find the corresponding style
-      var findIcon = function(olIcon, icons) {
-        var id = olIcon.getSrc();
-        for (var i = 0; i < icons.length; i++) {
-          var regex = new RegExp('/' + icons[i].id + '-24');
-          if (regex.test(id)) {
-            return icons[i];
-          }
+        if (onLastPoint) {
+          helpMsgId = 'draw_snap_last_point_';
         }
-        return icons[0];
-      };
-
-      var findIconSize = function(olIcon, iconSizes) {
-        var scale = olIcon.getScale();
-        for (var i = 0; i < iconSizes.length; i++) {
-          if (scale == iconSizes[i].scale) {
-            return iconSizes[i];
-          }
+        if (onFirstPoint) {
+          helpMsgId = 'draw_snap_first_point_';
         }
-        return iconSizes[0];
-      };
-
-      var findColor = function(olColor, colors) {
-        var rgb = ol.color.asString(olColor.slice(0, 3));
-        for (var i = 0; i < colors.length; i++) {
-          if (rgb == ol.color.asString(colors[i].fill)) {
-            return colors[i];
-          }
+      }
+      overlay.getElement().innerHTML = $translate.instant(helpMsgId + type);
+    };
+
+    // Display an help tooltip when modifying
+    var updateModifyHelpTooltip = function(overlay, type, onExistingVertex) {
+      if (!overlay) {
+        return;
+      }
+      var helpMsgId = 'modify_new_vertex_';
+      if (onExistingVertex) {
+        helpMsgId = 'modify_existing_vertex_';
+      }
+      overlay.getElement().innerHTML = $translate.instant(helpMsgId + type);
+    };
+
+    // Display an help tooltip when selecting
+    var updateSelectHelpTooltip = function(overlay, type) {
+      if (!overlay) {
+        return;
+      }
+      var helpMsgId = 'select_no_feature';
+      if (type) {
+        helpMsgId = 'select_feature_' + type;
+      }
+      overlay.getElement().innerHTML = $translate.instant(helpMsgId);
+    };
```

Comment:
The 3 update help tooltip functions could be refactored as all 3 functions contain quite some common code. Minor though.